StringIndexOutOfBoundsException: String index out of range: -1 when file is scanned that contains the 'class' keyword only in comments

## Summary

It looks like [this issue](https://github.com/societe-generale/arch-unit-maven-plugin/issues/49) triggered with the Maven plugin, is caused by a general problem:
```
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
        at java.base/java.lang.String.substring(String.java:1837)
        at com.societegenerale.commons.plugin.service.JavaFileParser.extractClassName(JavaFileParser.java:124)
        at com.societegenerale.commons.plugin.service.JavaFileParser.parse(JavaFileParser.java:44)
        at com.societegenerale.commons.plugin.service.ExcludedPathsPreProcessor.determineClassNames(ExcludedPathsPreProcessor.java:194)
        at com.societegenerale.commons.plugin.service.ExcludedPathsPreProcessor.processExcludedPaths(ExcludedPathsPreProcessor.java:65)
        at com.societegenerale.commons.plugin.service.RuleInvokerService.<init>(RuleInvokerService.java:49)
        at com.societegenerale.commons.plugin.maven.ArchUnitMojo.execute(ArchUnitMojo.java:94)
        ... 13 common frames omitted
```

I think the core module should handle this case gracefully, and provide means to debug it, even if may not directly lead a resolution of the issue triggered in the Maven plugin.

## Type of Issue
<!-- This issue is a -->
<!-- put an `x` the boxe that apply. -->
It is a : 
 - [x] bug
 - [ ] request
 - [ ] question regarding the documentation


## Current Behavior

Passing file content that contain `class` only in a comment, leads to the process being aborted, with no additional information to resolve the issue.

## Expected Behavior

The core module should provide constructive feedback, and proceed with the next file.

## Steps to Reproduce (for bugs)

See https://github.com/societe-generale/arch-unit-maven-plugin/issues/49

## Your Environment

See https://github.com/societe-generale/arch-unit-maven-plugin/issues/49

--- a/src/main/java/com/societegenerale/commons/plugin/service/JavaFileParser.java
+++ b/src/main/java/com/societegenerale/commons/plugin/service/JavaFileParser.java
@@ -115,12 +115,18 @@ class JavaFileParser
     @Nullable
     String extractClassName(final String javaFileContent, @Nonnull final Log logger)
     {
-        if (javaFileContent == null || javaFileContent.isEmpty() || !javaFileContent.contains(CLASS))
+        if (javaFileContent == null || javaFileContent.isEmpty())
         {
             return null;
         }
 
         final String commentsRemoved = removeComments(javaFileContent);
+        if (!commentsRemoved.contains(CLASS))
+        {
+            logger.warn("no class found in file with content: " + javaFileContent);
+            return null;
+        }
+
         final String classOffsetContent = commentsRemoved.substring(commentsRemoved.indexOf(CLASS));
 
         // split at the first "{"

--- a/src/test/java/com/societegenerale/commons/plugin/service/AbstractExcludePathTest.java
+++ b/src/test/java/com/societegenerale/commons/plugin/service/AbstractExcludePathTest.java
@@ -35,6 +35,13 @@ abstract class AbstractExcludePathTest
                                                             + "   public static class InnerClass{}"
                                                             + "}";
 
+    private static final String CONTENT_WITH_CLASS_IN_COMMENT = "/** \r\n"
+                                                            + "public class " + CLASS_NAME_WITH_FILE_COMMENT
+                                                            + "{ \r\n"
+                                                            + "   public static class InnerClass{}"
+                                                            + "}"
+                                                            + "*/\n";
+
     static final String PACKAGE_NAME         = AbstractExcludePathTest.class.getPackage().getName();
     static final String CONTENT_WITH_PACKAGE = "package " + PACKAGE_NAME + ";\r\n"
                                                + CONTENT_WITH_DEFAULT_PACKAGE;
@@ -42,6 +49,10 @@ abstract class AbstractExcludePathTest
     static final String CONTENT_WITH_FILE_COMMENT_AND_PACKAGE =
             fileComment + "package " + PACKAGE_NAME + ";\r\n"
             + CONTENT_WITH_FILE_COMMENT;
+
+    static final String CONTENT_WITH_CLASS_IN_COMMENT_AND_PACKAGE =
+            fileComment + "package " + PACKAGE_NAME + ";\r\n"
+            + CONTENT_WITH_CLASS_IN_COMMENT;
 
     private static final String PROJECT_BUILD_DIR_NAME = "target";
 

--- a/src/test/java/com/societegenerale/commons/plugin/service/JavaFileParserTest.java
+++ b/src/test/java/com/societegenerale/commons/plugin/service/JavaFileParserTest.java
@@ -111,7 +111,7 @@ public class JavaFileParserTest extends AbstractExcludePathTest
     }
 
     @Test
-    public void testExtractClassName_ClassInComments()
+    public void testExtractClassNameClassInComments()
     {
         final String className = javaFileParser.extractClassName(
                 AbstractExcludePathTest.CONTENT_WITH_CLASS_IN_COMMENT_AND_PACKAGE,

--- a/src/test/java/com/societegenerale/commons/plugin/service/JavaFileParserTest.java
+++ b/src/test/java/com/societegenerale/commons/plugin/service/JavaFileParserTest.java
@@ -111,6 +111,15 @@ public class JavaFileParserTest extends AbstractExcludePathTest
     }
 
     @Test
+    public void testExtractClassName_ClassInComments()
+    {
+        final String className = javaFileParser.extractClassName(
+                AbstractExcludePathTest.CONTENT_WITH_CLASS_IN_COMMENT_AND_PACKAGE,
+                getLogger());
+        assertNull(className);
+    }
+
+    @Test
     public void testRemoveComments()
     {
         final String blockComment = "/**\n\r"

